spec/server/proxyClientRequestToApiSpec.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+var proxyClientRequestToApiSpec = require("../../lib/server/proxyClientRequestToApi");
+
+describe("proxyClientRequestToApi", function() {
+
+    it("returns a middleware handler", function(){
+        var result = proxyClientRequestToApiSpec('localhost');
+        expect(typeof(result)).toBe("function");
+    });
+
+    describe("middleware handler", function() {
+        var apiHost = 'localhost';
+        var middlewareHandler = proxyClientRequestToApiSpec(apiHost);
+
+        it("should call next if header is not set", function(){
+            var next = jasmine.createSpy('next');
+            middlewareHandler({headers: {}},{},next);
+            expect(next).toHaveBeenCalled();
+        });
```

Comment:
Yes. I was waiting for the discussion above to settle before finalizing the code and writing the tests.